api/src/test/java/org/xwiki/filemanager/internal/reference/DefaultUniqueDocumentReferenceGeneratorTest.java

```diff
@@ -95,4 +114,27 @@ public void generate() throws Exception
         assertEquals(spaceReference, secondReference.getLastSpaceReference());
         assertEquals("foo2", secondReference.getName());
     }
+
+    @Test
+    public void namingStrategy() throws Exception {
```

Comment:
Usually when naming a test you start by mentioning the method you are testing. In here you could have `generateWithNamingStrategy`